storage/src/smt/mod.rs

```diff
@@ -1 +1,666 @@
+        let snapshot = self.db.snapshot();
+        let smt = get_smt!(self.db, &STAKER_TABLE, &prefix, &snapshot);
+
+        Ok(smt.root().clone())
+    }
+
+    async fn generate_sub_proof(&self, epoch: Epoch, addresses: Vec<Address>) -> Result<Proof> {
+        let prefix = SmtPrefixType::Epoch(epoch).as_prefix();
+        let snapshot = self.db.snapshot();
+        let keys = addresses
+            .into_iter()
+            .map(|k| SmtKeyEncode::Address(k).to_h256())
+            .collect::<Vec<H256>>();
+        let smt = get_smt!(self.db, &STAKER_TABLE, &prefix, &snapshot);
+
+        Ok(smt.merkle_proof(keys.clone())?.compile(keys)?.into())
+    }
+
+    async fn generate_top_proof(&self, epochs: Vec<Epoch>) -> Result<Proof> {
+        let prefix = SmtPrefixType::Top.as_prefix();
+        let snapshot = self.db.snapshot();
+        let keys = epochs
+            .into_iter()
+            .map(|k| SmtKeyEncode::Epoch(k).to_h256())
+            .collect::<Vec<H256>>();
+        let smt = get_smt!(self.db, &STAKER_TABLE, &prefix, &snapshot);
+
+        Ok(smt.merkle_proof(keys.clone())?.compile(keys)?.into())
+    }
+}
+
+/// Delegator SMTs
+/// Each smt stores one staker's delegation information.
+/// For sub smt, the key is the delegator address, the value is the amount of
+/// delegation. For top smt, the key is the epoch, the value is the root of sub
+/// smt.      
+///                               Staker 1 Root
+///                  /                                     \
+///            epoch 1 root       ...                 epoch 2 root
+///      /          |           \               /          |          \
+/// delegator1  delegator2  delegator3      delegator1  delegator2  delegator4
+///  amount1     amount2     amount3        amount1     amount2     amount4
+///                                     .
+///                                     .
+///                                     .
+///                               Staker k Root
+///                  /                                     \
+///            epoch 1 root       ...                 epoch 2 root
+///      /          |            \              /          |          \
+/// delegator1  delegator2  delegator3      delegator1  delegator2  delegator4
+///  amount1     amount2     amount3         amount1     amount2     amount4
+///
+///  Column family prefix in RocksDB: 'delegator' --> "delegator_branch" and
+///  "delegator_leaf" Tree prefix in Column family: epoch.to_le_bytes() +
+///  stake_address.to_fixed_bytes()
+///
+///  Top SMT
+///      key: 'top_smt'.as_slice() + staker_address.to_fixed_bytes()
+///      value: root(H256)
+///
+///  Sub SMT
+///     key: delegator_address(H160).to_fixed_bytes() + [0u8; 12]
+///     value: amount(u128).to_fixed_bytes() + [0u8; 16]
+#[async_trait]
+impl DelegateSmtStorage for SmtManager {
+    async fn insert(
+        &self,
+        epoch: Epoch,
+        delegators: HashMap<Staker, Vec<(Delegator, Amount)>>,
```

Comment:
```suggestion
        delegators: Vec<Staker, Vec<(Delegator, Amount)>>,
```

The merge operation should be done in this function.